### PR TITLE
fix resolve redirect to pass all original arguments

### PR DESCRIPTION
**Pull request by pvanderlinden**
Fix for https://github.com/kennethreitz/requests/issues/2503
It is still a bit inconsistent but as mentioned in the issue, but it otherwise will break the current public api. 

This makes it possible for non urllib3 adapters to accept extra arguments.

### CODE REVIEW

**Comment by Lukasa** [comment]
I think I'd like the `kwargs` parameter in `resolve_redirects` to have a different name: `adapter_kwargs`, maybe?

**Comment by pvanderlinden** [comment]
I have changed the parameter.

**Comment by Lukasa** [comment]
LGTM. The one thing I'm thinking about is whether we want to allow for forward compatibility by actually storing these kwargs in a dictionary for resolve_redirects, but I haven't decided yet. I'll let @sigmavirus24 make a call here.

File: requests/sessions.py [hunk 1]
```diff
@@ -560,10 +561,6 @@ def send(self, request, **kwargs):
         # Set up variables needed for resolve_redirects and dispatching of hooks
         allow_redirects = kwargs.pop('allow_redirects', True)
         stream = kwargs.get('stream')
```
Comment (sigmavirus24) [inline requests/sessions.py]: Does stream need to be extracted here?

**Comment by sigmavirus24** [comment]
> The one thing I'm thinking about is whether we want to allow for forward compatibility by actually storing these kwargs in a dictionary for resolve_redirects

Could you expand on that @Lukasa? I'm not quite certain what you mean.

**Comment by pvanderlinden** [comment]
Any thoughts? I would like to get this merged as it is preventing me from adding a feature to my adapter (making it open source soon).

**Comment by Lukasa** [comment]
Sorry, what I meant was I'm wondering whether we should, rather than using `**kwargs` the whole way through, save the initial `**kwargs` in a dictionary which is then passed _as a dictionary_. That way, if we ever feel the need to add `**kwargs` to any of these functions later we could.

I don't think it's a big deal though.

**Comment by sigmavirus24** [comment]
I'm okay with this if you are @Lukasa.

**Comment by Lukasa** [comment]
Sure, why not. =)

**Comment by pvanderlinden** [comment]
Thanks!

**Pull request merged**